nchelpers/__init__.py

```diff
@@ -371,41 +372,47 @@ class UnifiedMetadata(object):
-            # at least by CMIP5 notions
-            'project': {
-                'CMIP3': 'project_id',
-                'CMIP5': 'project_id',
+        _aliases_by_project_id = {
+            'CMIP3': {
+                # Original aliases - some mangle the terminology somewhat,
+                'project': 'project_id',
+                'institution': 'institute',
+                'model': 'source',
+                'emissions': 'experiment_id',
+                'run': 'realization',
+                # Better aliases - adhere to CMIP5 terminology
+                'institute': 'institute',
+                'experiment': 'experiment_id',
+                'ensemble_member': 'realization',
             },
-            'institution': {
-                'CMIP3': 'institute',
-                'CMIP5': 'institute_id',
-            },
-            'model': {
-                'CMIP3': 'source',
-                'CMIP5': 'gcm.model_id',
-            },
-            'emissions': {
-                'CMIP3': 'experiment_id',
-                'CMIP5': 'gcm.experiment_id',
-            },
-            'run': {
-                'CMIP3': 'realization',
-                'CMIP5': 'ensemble_member',  # uses prefixed values
-            },
-            # Better aliases - adhere to CMIP5 terminology
-            'institute': {
-                'CMIP3': 'institute',
-                'CMIP5': 'institute_id',
-            },
-            'experiment': {
-                'CMIP3': 'experiment_id',
-                'CMIP5': 'gcm.experiment_id',
+
+            'CMIP5': {
+                # Original aliases - some mangle the terminology somewhat,
+                'project': 'project_id',
+                'institution': 'institute_id',
+                'model': 'gcm.model_id',
+                'emissions': 'gcm.experiment_id',
+                'run': 'ensemble_member',  # uses prefixed values
+                # Better aliases - adhere to CMIP5 terminology
+                'institute': 'institute_id',
+                'experiment': 'gcm.experiment_id',
+                'ensemble_member': 'ensemble_member',  # uses prefixed values
             },
-            'ensemble_member': {
-                'CMIP3': 'realization',
-                'CMIP5': 'ensemble_member',  # uses prefixed values
+
+            # CAUTION: This mapping uses attribute names (for both metadata.*
+            # and mapped attributes) that apply to CMIP datasets. 'other' type
+            # datasets are not necessarily so. This is a minimal temporary
+            # solution for a priority project.
+            'other': {
+                # Original aliases - some mangle the terminology somewhat,
+                'project': 'project_id',
+                'institution': 'institute_id',
+                'model': 'model_id',
+                'emissions': 'experiment_id',
+                'run': 'run',  # bogus attribute to replace ensemble_member
```

Comment:
Yeah, this will work for now (even if it feels kind of icky). Long term solution is to make `run` an optional attribute for files that are not CMIP/climate model derived?

Reply:
Yes, more or less. The specifics will depend on what we add to the PCIC metadata standard. 'Run' is a term from the CMIP/climate model arena. Here's my [preferred solution](https://github.com/pacificclimate/modelmeta/issues/52) to the whole issue.